File: composeApp/src/androidMain/kotlin/com/kakapo/oakane/presentation/feature/wallet/component/dialog/WalletDialogView.kt
package com.kakapo.oakane.presentation.feature.wallet.component.dialog

import androidx.compose.material3.BasicAlertDialog
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.runtime.Composable
import com.kakapo.oakane.presentation.ui.component.dialog.DeleteContentDialogView
import com.kakapo.oakane.presentation.viewModel.wallet.WalletDialogContent
import com.kakapo.oakane.presentation.viewModel.wallet.WalletEvent
import com.kakapo.oakane.presentation.viewModel.wallet.WalletState

@OptIn(ExperimentalMaterial3Api::class)
@Composable
internal fun WalletDialogView(uiState: WalletState, onEvent: (WalletEvent) -> Unit) {
    BasicAlertDialog(
        onDismissRequest = { onEvent.invoke(WalletEvent.ShowDialog(shown = false)) },
        content = { DialogContentView(uiState, onEvent) }
    )
}

@Composable
private fun DialogContentView(uiState: WalletState, onEvent: (WalletEvent) -> Unit) {
    val deleteTitle = "Are you sure want to delete this wallet?"
    val deleteSubtitle =
        "This action can't be undone and will delete all transactions in this wallet"
    Surface(shape = MaterialTheme.shapes.large) {
        when (uiState.dialogContent) {
            WalletDialogContent.DeleteWallet -> DeleteContentDialogView(
                title = deleteTitle,
                subtitle = deleteSubtitle,
                onDismiss = { onEvent.invoke(WalletEvent.ShowDialog(shown = false)) },
                onConfirm = { onEvent.invoke(WalletEvent.ConfirmDelete) }
            )

            WalletDialogContent.MoveBalance -> MoveBalanceDialogContentView(uiState, onEvent)
        }
    }
}

File: iosApp/iosApp/screen/wallet/component/WalletDialogView.swift
import SwiftUI
import Shared

struct WalletDialogView: View {
    
    let uiState: WalletState
    let onEvent: (WalletEvent) -> Void
    
    private let title = "Are you sure want to delete this wallet?"
    private let subtitle = "This action can't be undone and will delete all transactions in this wallet"
    private let dismissEvent: WalletEvent = .ShowDialog(content: .moveBalance, shown: false)
    private let confirmEvent: WalletEvent = .ConfirmDelete()
    
    var body: some View {
        PopUpDialog(onDismiss: { _ in onEvent(dismissEvent) }) {
            DeleteContentDialogView(
                title: title,
                subtitle: subtitle,
                onDismiss: { onEvent(dismissEvent)},
                onConfirm: { onEvent(confirmEvent) }
            )
        }
    }
}

#Preview {
    let defaultValue = WalletState.companion.default()
    WalletDialogView(uiState: defaultValue, onEvent: { _ in })
}
